smart-contract-verifier/smart-contract-verifier-server/src/services/sourcify_verifier.rs

```diff
@@ -1,17 +1,52 @@
+use crate::{
+    metrics,
+    settings::SourcifySettings,
+    types::{VerifyResponseWrapper, VerifyViaSourcifyRequestWrapper},
+};
+use smart_contract_verifier::{sourcify, sourcify::Error, SourcifyApiClient};
 use smart_contract_verifier_proto::blockscout::smart_contract_verifier::v1::{
     sourcify_verifier_server::SourcifyVerifier, VerifyResponse, VerifyViaSourcifyRequest,
 };
+use std::sync::Arc;
 use tonic::{Request, Response, Status};
 
-#[derive(Default)]
-pub struct SourcifyVerifierService {}
+pub struct SourcifyVerifierService {
+    client: Arc<SourcifyApiClient>,
+}
+
+impl SourcifyVerifierService {
+    pub fn new(settings: SourcifySettings) -> anyhow::Result<Self> {
+        let client = SourcifyApiClient::new(
+            settings.api_url,
+            settings.request_timeout,
+            settings.verification_attempts,
+        )
+        .expect("failed to build sourcify client");
```

Comment:
no extensions

Reply:
Added in https://github.com/blockscout/blockscout-rs/pull/226/commits/e3652d887b34462822b79b9e7739ade9a9277736